crypto/merklekeystore/keystore_test.go

```diff
@@ -17,176 +17,185 @@
+	signer = generateTestSigner(8, 20, 10, a)
+	defer signer.keyStore.store.Close()
+	a.Equal(length(signer, a), 2)
 
-	s, err = New(10, 21, 10, crypto.Ed25519Type)
-	a.NoError(err)
-	a.Equal(len(s.SignatureAlgorithms), 2)
+	signer = generateTestSigner(10, 21, 10, a)
+	defer signer.keyStore.store.Close()
+	a.Equal(length(signer, a), 2)
 
-	s, err = New(10, 20, 10, crypto.Ed25519Type)
-	a.NoError(err)
-	a.Equal(len(s.SignatureAlgorithms), 2)
+	signer = generateTestSigner(10, 20, 10, a)
+	defer signer.keyStore.store.Close()
+	a.Equal(length(signer, a), 2)
 
-	s, err = New(11, 20, 10, crypto.Ed25519Type)
-	a.NoError(err)
-	a.Equal(len(s.SignatureAlgorithms), 1)
+	signer = generateTestSigner(11, 20, 10, a)
+	defer signer.keyStore.store.Close()
+	a.Equal(length(signer, a), 1)
 
-	s, err = New(11, 19, 10, crypto.Ed25519Type)
-	a.NoError(err)
-	a.Equal(0, len(s.SignatureAlgorithms))
+	signer = generateTestSigner(11, 19, 10, a)
+	defer signer.keyStore.store.Close()
+	a.Equal(0, length(signer, a))
 	_, err = signer.Sign(genHashableForTest(), 2)
 	a.Error(err)
 }
 func TestEmptyVerifier(t *testing.T) {
 	a := require.New(t)
 
-	signer, err := New(8, 9, 5, crypto.Ed25519Type)
-	a.NoError(err)
+	signer := generateTestSigner(8, 9, 5, a)
+	defer signer.keyStore.store.Close()
 	a.NotEqual(*signer.GetVerifier(), Verifier{})
-
 }
 func TestEmptySigner(t *testing.T) {
 	a := require.New(t)
+	var err error
 
 	h := genHashableForTest()
-	signer, err := New(8, 9, 5, crypto.Ed25519Type)
+	signer := generateTestSigner(8, 9, 5, a)
 	a.NoError(err)
-	a.Equal(0, len(signer.SignatureAlgorithms))
+	a.Equal(0, length(signer, a))
 
 	_, err = signer.Sign(h, 8)
 	a.Error(err)
 
 	_, err = signer.Sign(h, 9)
 	a.Error(err)
 
-	_, err = signer.Trim(10)
-	a.Error(err)
+	// should Trim return error when trying to trim above its last valid? or just drop what relevant keys he finds as it is currently?
+	//_, err = signer.Trim(10)
+	//a.Error(err)
 }
 
-func TestDisposableKeyPositions(t *testing.T) {
-	a := require.New(t)
-	signer, err := New(0, 100, 1, crypto.Ed25519Type)
-	a.NoError(err)
-
-	for i := uint64(1); i < 100; i++ {
-		pos, err := signer.getArrayIndex(i)
-		a.NoError(err, i)
-		a.Equal(i-1, pos)
-	}
-
-	_, err = signer.getArrayIndex(101)
-	a.Error(err)
-
-	signer, err = New(1000, 1100, 1, crypto.Ed25519Type)
-	a.NoError(err)
-
-	for i := uint64(1000); i < 1100; i++ {
-		pos, err := signer.getArrayIndex(i)
-		a.NoError(err, i)
-		a.Equal(i-1000, pos)
-	}
-
-	_, err = signer.getArrayIndex(999)
-	a.Error(err)
-
-	signer, err = New(1000, 1100, 101, crypto.Ed25519Type)
-	a.NoError(err)
-
-	indices := make([]uint64, 0)
-	for i := uint64(1000); i <= 1100; i++ {
-		if i%101 == 0 {
-			indices = append(indices, i)
-			continue
-		}
-		_, err := signer.getArrayIndex(i)
-		a.Error(err, i)
-	}
-
-	for index, round := range indices {
-		pos, err := signer.getArrayIndex(round)
-		a.NoError(err)
-		a.Equal(uint64(index), pos)
-	}
-}
+//func TestDisposableKeyPositions(t *testing.T) {
```

Comment:
we should keep those tests.

Reply:
Keys are no longer stored by index and offset by arrayBase so what makes this test useful?